### PR TITLE
House Cleaning Part 3 - Bye Bye Target

### LINKED ISSUE #822: Dash code Problem
When you use DashToLocation or DashToTarget
the animation was Dashing, but actually it was walking to there, and speed(dash) was for animation, actual spot like i said, it was walking to target (you can see the problem when you CancelDash earlier)
so It need to fix:
1. Dash to target should not be walking to there
2. If it can't fix, then use Walking Speed = Dash Speed, when it reach last position, CancelDash.

### PR DESCRIPTION
**Pull request by lzardy**
* Target is dead, Vector2 is supreme.
* Because of the original Dash packet relying on Target, I had to rewrite all of the functionality for Dash.
* Scripting API now supports targeted and positional dashes separately.
* Added partial LeeSin Q script to demonstrate targeted dashes.
* Moved movement functionality from GameObject to AttackableUnit.
* All Target based functionality is either replaced by Vector2 or a GameObject/surtype of GameObject.
* AttackableUnits support positional based dashes, and ObjAIBases support target based dashes.
* Deleted more unused packets.
* Target WithinRange function replaced by Extensions.IsVectorWithinRange
* All GameObject surtypes spawn with a Vector2 position instead of separate x and y coordinates.
* Target GetDistanceTo replaced with Vector2.Distance/Squared.
* GetPosition3D function added to GameObject to replace unclean Vector3 initializations.
* Removed ambiguous StopChampionMovement and replaced with override StopMovement in ObjAiBase.
* Rearranged AttackableUnit by ordering by least to greatest complexity of features.

Resolves #822, #721, #545
Progress #962, #757

### CODE REVIEW

File: GameServerLib/GameObjects/Other/CollisionHandler.cs [hunk 1]
```diff
@@ -106,7 +106,7 @@ public void Update()
                     continue;
                 }
 
-                if (!_map.NavigationGrid.IsWalkable(obj.X, obj.Y))
+                if (!_map.NavigationGrid.IsWalkable(obj.Position.X, obj.Position.Y))
```
Comment (danil179) [inline GameServerLib/GameObjects/Other/CollisionHandler.cs:109]: IsWalkable still gets X,Y

**Review by danil179 — COMMENTED** [review]
Some comments:
1. Why DashToLocation is in IAttackableUnit and DashToTarget is in ObjAIBase?
2. There are still some places with castings between vectors and X,Y coordinates. (ignore this, I'll open an issue about that)

I still need to test if that runs correctly.

File: GameServerLib/GameObjects/Other/CollisionHandler.cs (hunk 1 above)
Comment (lzardy) [reply GameServerLib/GameObjects/Other/CollisionHandler.cs:109]: AttackableUnits can move, but can't target, hence the separation between dashing to a location vs a target object.

Comment (HarmfulBreeze) [reply GameServerLib/GameObjects/Other/CollisionHandler.cs:109]: I'm kind of out of the loop, but IIRC an inhibitor is an ``AttackableUnit``, so I'm not sure why an ``AttackableUnit`` should be able to move...

Comment (lzardy) [reply GameServerLib/GameObjects/Other/CollisionHandler.cs:109]: Although AttackableUnits can move, this doesn't mean they will move by themselves. The only instance where they will move is when an attack forces them to dash. Buildings of course will override this.

**Review by danil179 — COMMENTED** [review]
Seems fine. I have problems with events called 'onCollide' and other events starting with uppercase 'OnDie'. I think we should change all to start with uppercase or something similar. Probably for other PR.

I'll test and see if that works.

**Review by danil179 — APPROVED** [review]
Tested and everything seems to work well with Ezreal + mininos. There are 2 problems regarding vision and AI:
1. Sometimes enemy minions are sort of teleporting or showed in the wrong position if you don't have vision on, then appearing when spotted in a new location.
2. Minions AI sometimes cause the minions to stop and "forget" about attacking/walking to the next buildings.

**Comment by danil179** [comment]
Resolve #955

**Pull request merged**